crates/mpz-common/src/queue/round_robin.rs

```diff
@@ -0,0 +1,134 @@
+use std::{collections::VecDeque, mem};
+
+use futures::Future;
+
+use crate::{
+    queue::{Queue, Task},
+    ContextError,
+};
+
+/// A queue that load balances tasks in a round-robin fashion.
+pub struct RRQueue<'a, L, R> {
+    lanes: &'a mut [L],
+    queue: VecDeque<Box<dyn Task<L, R>>>,
+}
+
+impl<'a, L, R> RRQueue<'a, L, R> {
+    /// Creates a new round-robin queue.
+    pub fn new(lanes: &'a mut [L]) -> Self {
+        Self {
+            lanes,
+            queue: VecDeque::new(),
+        }
+    }
+}
+
+impl<'a, L, R> Queue<L, R> for RRQueue<'a, L, R>
+where
+    L: Send,
+    R: Send,
+{
+    fn push<F>(&mut self, f: F)
+    where
+        F: Task<L, R>,
+    {
+        self.queue.push_back(Box::new(f));
+    }
+
+    fn wait(&mut self) -> impl Future<Output = Result<Vec<R>, ContextError>> + Send {
+        let lane_count = self.lanes.len();
+        let task_count = self.queue.len();
+
+        let mut lanes: Vec<_> = (0..lane_count)
+            .map(|_| Vec::with_capacity((task_count / lane_count) + (task_count % lane_count)))
```

Comment:
```suggestion
            .map(|_| Vec::with_capacity((task_count / lane_count) + 1))
```
One more should be enough, right?

Reply:
should be + (task_count % lane_count != 0)